scripts/github-repos/github-repos.rb

```diff
@@ -7,220 +7,408 @@
+      end
+      childteam_id = childteam['id']
+      @childteams[team].each do |username|
+        begin
+          @client.add_team_member(childteam_id, username)
+        rescue Octokit::UnprocessableEntity
+          # save the incorrect username
+          failed_username << username
         end
+      end
     end
 
-    def create_repos
-        print_error "csv file, base filename, template repo name, semester prefix, students team name, and gsi team name needed." unless valid?
-        @childteams.each_key do |team|
-            begin
-                team_id = @client.team_by_name(@orgname, %Q{#{@semester}-#{team}})['id']
-            rescue Octokit::NotFound
-                print_error "Students teams information mismatched."
-            end
-            gsiteam_id = @client.team_by_name(@orgname, @gsiteam)['id']
-            new_repo_name = %Q{#{@semester}-#{@base_filename}-#{team}}
-            if !@client.repository? %Q{#{@orgname}/#{new_repo_name}}
-                begin
-                    new_repo = @client.create_repository_from_template(%Q{#{@orgname}/#{@template}}, new_repo_name, 
-                        {owner: @orgname, private: true})
-                rescue Octokit::NotFound
-                    print_error "Template not found."
-                end
-                @client.add_team_repository(team_id, new_repo['full_name'], {permission: 'push'})
-                @client.add_team_repository(gsiteam_id, new_repo['full_name'], {permission: 'admin'})
-            end
+    # print the usernames that is failed to add to child team
+    if !failed_username.empty?
+      puts 'Failed to add these usernames to child team:'
+      puts failed_username
+    end
+  end
+
+  def indiv_repos
+    print_error "orgname, student team name, base filename, template repo name, semester prefix, and gsi team name needed." unless repos_valid?
+    
+    # Looking for the STUDENTTEAM in the org, see if it is exist.
+    begin
+      parentteam_id = @client.team_by_name(@orgname, to_slug(@parentteam))['id']
+    rescue Octokit::NotFound
+      print_error "Please make sure students team is created."
+    end
+
+    gsiteam_id =  @client.team_by_name(@orgname, to_slug(@gsiteam)).id
+    self_user_name = @client.user.login
+    @client.team_members(parentteam_id).each do |mem|
+      if self_user_name != mem.login
+        new_repo_name = %Q{#{@semester}-#{mem.login}-#{@base_filename}}
+        if !@client.repository? %Q{#{@orgname}/#{new_repo_name}}
+          begin
+            new_repo = @client.create_repository_from_template(%Q{#{@orgname}/#{@template}}, new_repo_name, 
+              {owner: @orgname, private: true})
+          rescue Octokit::NotFound
+            print_error "Template not found."
+          end
+          @client.add_collab(new_repo['full_name'], mem.login)
+          @client.add_team_repository(gsiteam_id, new_repo['full_name'], {permission: 'admin'})

```

Comment:
I think it's best to move this outside the if-statement started on line 287 -- that way, if something happens that prevents the collaborators from being added, we can re-run the script and they can get added to an existing repository the next time.

(I ran into this just now with chips 4.6 -- for some reason a student didn't end up assigned to their repo and it needed a re-run)